tiledb/sm/filter/bit_width_reduction_filter.cc

```diff
@@ -86,6 +86,11 @@ BitWidthReductionFilter::BitWidthReductionFilter()
   max_window_size_ = 256;
 }
 
+BitWidthReductionFilter::BitWidthReductionFilter(uint32_t max_window_size)
```

Comment:
This is the right C.41 constructor for this particular filter, but it's only called in tests, not in `deserialize`.

This PR should eliminate `deserialize_impl` in this class, which will force the filter `deserialize` function to use this constructor.